subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/InputProperties.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.changedetection.state;
+
+import org.gradle.api.GradleException;
+
+import java.io.File;
+import java.io.NotSerializableException;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.TreeMap;
+
+import static java.lang.String.format;
+
+public abstract class InputProperties {
+    public static Map<String, InputProperty> process(Map<String, Object> inputProperties) {
+        SortedMap<String, InputProperty> result = new TreeMap<String, InputProperty>(new HashMap<String, InputProperty>(inputProperties.size()));
```

Comment:
Couldn't we just use a `Map` data type instead of `SortedMap`? What's the driver for using `SortedMap`?